mlrun/datastore/model_provider/openai_provider.py

```diff
@@ -106,15 +108,23 @@ def _get_messages_parameter(
             ]
         else:
             raise mlrun.errors.MLRunInvalidArgumentError(
-                "must provide 'messages' or 'prompt' to invoke"
+                "must provide 'messages' or 'prompt' to invoke an OpenAIProvider"
             )
         return messages, invoke_kwargs
 
-    def invoke(self, prompt: Optional[str] = None, **invoke_kwargs) -> str:
+    def invoke(
+        self,
+        prompt: Optional[str] = None,
+        messages: Optional[list[dict]] = None,
+        as_str: bool = False,
+        **invoke_kwargs,
+    ) -> Optional[Union[str, T]]:
```

Comment:
please add docs... I think we can delete the `prompt` but if not please explain that you treat to it as a user prompt.